app/models/article.rb

```diff
@@ -417,6 +419,16 @@ def update_score
                    spaminess_rating: BlackBox.calculate_spaminess(self))
   end
 
+  def top_comments_cache_name
+    "article-#{id}-#{last_comment_at&.rfc3339}/top_comments"
+  end
+
+  def cached_top_comments
+    Rails.cache.fetch(top_comments_cache_name, expires_in: 24.hours) do
```

Comment:
No, that all makes sense: I have **no idea** how this will perform TBH. Here's an `explain analyze` of the query to get top comments:

```
Sort  (cost=444.67..444.67 rows=12 width=944) (actual time=3.646..3.646 rows=2 loops=1)
  Sort Key: comments.score
  Sort Method: quicksort  Memory: 28kB
  ->  Bitmap Heap Scan on comments  (cost=402.83..444.62 rows=12 width=944) (actual time=3.614..3.625 rows=2 loops=1)
        Recheck Cond: ((commentable_id = ANY ('{303199,302536,300600,302629,302747,302603,302462,302630,303213,302601,302461,302816,303266,302688,303222,302483,302634,303080,301184,301941,302616,302642,303359,302328,302769,302454,302751,302480,302740,302679,302622,302495,302833,302785,302826}'::integer[])) AND ((commentable_type)::text = 'Article'::text) AND (score > 10))
        Filter: (ancestry IS NULL)
        Heap Blocks: exact=2
        ->  BitmapAnd  (cost=402.83..402.83 rows=21 width=0) (actual time=3.586..3.586 rows=0 loops=1)
              ->  Bitmap Index Scan on index_comments_on_commentable_id_and_commentable_type  (cost=0.00..74.08 rows=547 width=0) (actual time=0.167..0.167 rows=187 loops=1)
                    Index Cond: ((commentable_id = ANY ('{303199,302536,300600,302629,302747,302603,302462,302630,303213,302601,302461,302816,303266,302688,303222,302483,302634,303080,301184,301941,302616,302642,303359,302328,302769,302454,302751,302480,302740,302679,302622,302495,302833,302785,302826}'::integer[])) AND ((commentable_type)::text = 'Article'::text))
              ->  Bitmap Index Scan on index_comments_on_score  (cost=0.00..328.70 rows=15076 width=0) (actual time=3.351..3.351 rows=19210 loops=1)
                    Index Cond: (score > 10)
Planning Time: 0.878 ms
Execution Time: 3.703 ms
```

It doesn't look too bad? I think the best course of action right now would be to leave out the Rails caching of top comments, and add it in if we see performance is lagging.

Reply:
Yeah, 3ms is probably less than the roundtrip to Redis :-) Let's test it out in production :D